buildspec.yml

```diff
@@ -2,16 +2,16 @@ version: 0.2
 
 env:
   variables:
-    FRAMEWORK_VERSION: '1.10.2'
+    FRAMEWORK_VERSION: '2.0.1'
     EIA_FRAMEWORK_VERSION: '1.3.1'
     CPU_INSTANCE_TYPE: 'ml.c4.xlarge'
-    GPU_INSTANCE_TYPE: 'ml.p3.8xlarge'
+    GPU_INSTANCE_TYPE: 'ml.g4dn.xlarge'
```

Comment:
This host only have 1 gpu. We might need a host have 2 gpus for some test cases.

Reply:
Updated the GPU instance type to `g4dn.12xlarge`. This instance has 4 GPUs. Please note that we need to pass the environment variable `NCCL_SHM_DISABLE=1` while creating the SageMaker endpoints to avoid this error:
```
NCCL Error 2: unhandled system error
Traceback (most recent call last):
  File "/opt/conda/lib/python3.8/site-packages/sagemaker_inference/transformer.py", line 142, in transform
    result = self._run_handler_function(
  File "/opt/conda/lib/python3.8/site-packages/sagemaker_inference/transformer.py", line 276, in _run_handler_function
    result = func(*argv_context)
  File "/opt/conda/lib/python3.8/site-packages/sagemaker_inference/transformer.py", line 260, in _default_transform_fn
    prediction = self._run_handler_function(self._predict_fn, *(data, model))
  File "/opt/conda/lib/python3.8/site-packages/sagemaker_inference/transformer.py", line 272, in _run_handler_function
    result = func(*argv)
  File "/opt/conda/lib/python3.8/site-packages/sagemaker_pytorch_serving_container/default_pytorch_inference_handler.py", line 125, in default_predict_fn
    output = model(input_data)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/parallel/data_parallel.py", line 167, in forward
    replicas = self.replicate(self.module, self.device_ids[:len(inputs)])
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/parallel/data_parallel.py", line 172, in replicate
    return replicate(module, device_ids, not torch.is_grad_enabled())
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/parallel/replicate.py", line 91, in replicate
    param_copies = _broadcast_coalesced_reshape(params, devices, detach)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/parallel/replicate.py", line 67, in _broadcast_coalesced_reshape
    return comm.broadcast_coalesced(tensors, devices)
  File "/opt/conda/lib/python3.8/site-packages/torch/nn/parallel/comm.py", line 58, in broadcast_coalesced
    return torch._C._broadcast_coalesced(tensors, devices, buffer_size)
RuntimeError: NCCL Error 2: unhandled system error
```
Please note that this error is specifically related to `torch.nn.DataParallel` (https://github.com/pytorch/pytorch/issues/73775) which is used while creating the MNIST model. It causes the worker to die and leads to error code `500`.